finetuner/tuner/callback/training_checkpoint.py

```diff
@@ -0,0 +1,142 @@
+import os
+import pickle
+from typing import TYPE_CHECKING
+
+import keras
+import paddle
+import torch
+from jina.logging.logger import JinaLogger
+
+from finetuner.helper import get_framework
+
+from .base import BaseCallback
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class TrainingCheckpoint(BaseCallback):
+    """
+    Callback to save model at every epoch or the last K epochs
+    `TrainingModelCheckpoint` is used in conjunction with training
+    using `finetuner.fit()`
+    """
+
+    def __init__(self, save_dir: str, last_k_epochs: int = None):
+        """
+        :param save_dir: string or `PathLike`, path to save the model file.
+        :param last_k_epochs: this parameter is an integer. It allows you yo save
+            only at the last k epochs and not on every epoch.
+            For example if `last_k_epochs = 3` and the total number of epochs is 10
+            we'll save only on epochs 8,9 and 10.
```

Comment:
This is actually not how this feature should be implemented.

The point of this callback is to save the intermediate progress of training, in case something happens and the training is interrupted. There is no use case for saving only last 3 epochs.

What you want instead, is to save the _most recent_ 3 epochs. For example:
```
epoch 5 -> saved 5, 4, 3
epoch 6 -> saved 6, 5, 4 (here saved epoch 3 is deleted)
epoch 7 -> saved 7, 6, 5 (here saved epoch 4 is deleted)
```
And by default I would only keep the latest epoch (which is what most users would want)

Reply:
Now I get it